From b13ae6f29a1ff8892aab058158fe044de4845652 Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Mon, 7 Aug 2023 11:04:35 +0200
Subject: [PATCH 07/15] wintrust: Do not crash in CryptCATEnumerateMember when
 trying to enumerate members of an invalid file.

TODO: Check if this is correct (doesn't crash on Windows + check returned error).
---
 dlls/wintrust/crypt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/wintrust/crypt.c b/dlls/wintrust/crypt.c
index 11111111111..11111111111 100644
--- a/dlls/wintrust/crypt.c
+++ b/dlls/wintrust/crypt.c
@@ -659,7 +659,7 @@ CRYPTCATMEMBER * WINAPI CryptCATEnumerateMember(HANDLE hCatalog, CRYPTCATMEMBER
 
     TRACE("%p, %p\n", hCatalog, prev);
 
-    if (!hCatalog || hCatalog == INVALID_HANDLE_VALUE || cc->magic != CRYPTCAT_MAGIC)
+    if (!hCatalog || hCatalog == INVALID_HANDLE_VALUE || cc->magic != CRYPTCAT_MAGIC || !cc->inner)
     {
         SetLastError(ERROR_INVALID_PARAMETER);
         return NULL;
-- 
0.0.0

